ast-grep/rules/value-variant.yml

```
@@ -0,0 +1,17 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/ast-grep/ast-grep/main/schemas/rule.json
+
+id: value-variant-direct
+language: rust
+severity: error
+message: "Use one of the value variant helpers"
+
+ignores:
+  - "crates/nu-protocol/src/value/*.rs"
+
+rule:
+  kind: struct_expression
+  has:
+    kind: scoped_type_identifier
+    has:
+      kind: identifier
+      pattern: Value
```

```suggestion
rule:
  kind: struct_expression > scoped_type_identifier > identifier
  pattern: Value
```
After v0.39.1 it can be much more [concise!](https://ast-grep.github.io/guide/rule-config/atomic-rule.html#esquery-style-kind)

---

maybe we should hold off on this since it's experimental? not to pressed about it either way, though

---

yeah probably